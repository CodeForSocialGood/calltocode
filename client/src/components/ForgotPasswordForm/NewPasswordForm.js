import React, { Component } from 'react'
import { connect } from 'react-redux'
import { Field, reduxForm } from 'redux-form'
import PropTypes from 'prop-types'

import SignupValidator from '../SignupForm/SignupValidator'
import ValidationPopup from '../SignupForm/ValidationPopup'
import styles from './ForgotPasswordForm.scss'

class NewPasswordForm extends Component {
  renderPassword (field) {
    const passClasses = `${styles.inputPassword} 
                         ${field.meta.pristine ? '' : (field.meta.error ? styles.error : styles.valid)}`
    return (
      <div className={ styles.inputPasswordContainer }>
        <input
          className={ passClasses }
          placeholder="New Password"
          type="password"
          {...field.input} />
        <Field
          name="popup"
          component={ValidationPopup}
          active={field.meta.active}
          error={field.meta.error} />
      </div>
    )
  }

  render () {
    const { handleSubmit } = this.props

    return (
      <form className={styles.form} onSubmit={handleSubmit}>
        <h1 className={styles.title}>New Password</h1>
        <h3>Set your new password</h3>
        <Field
          name="password"
          component={this.renderPassword}
          validate={SignupValidator.validatePassword} />
        <button className={styles.buttonSubmit} type="submit">
          Change Password
        </button>
      </form>
    )
  }
}

NewPasswordForm.propTypes = {
  changePassword: PropTypes.func,
  handleSubmit: PropTypes.func
}

const NewPasswordFormRedux = reduxForm({
  form: 'ForgotPasswordForm', // <------ same form name
  destroyOnUnmount: false, // <------ preserve form data
  forceUnregisterOnUnmount: true // <------ unregister fields on unmount
})(NewPasswordForm)

export default connect(null)(NewPasswordFormRedux)
