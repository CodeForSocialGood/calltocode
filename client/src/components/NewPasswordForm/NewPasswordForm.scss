@import '../../scss/partials/partials';
@import '../../scss/elements/form';
@import '../LoginForm/LoginForm';

.form {
  @extend %form;
}

h1 {
  @extend %formTitle;
  grid-column: 1 / 1;
  grid-row: 1;
}

.inputPassword {
  @extend %fieldText;
  grid-column: 1 / 1;
  grid-row: 2;
  border: none;
  border-bottom: 1px solid $inputsBorderColor;
}

.buttonSubmit {
  @extend %buttonSubmit;
  grid-column: 1 / 1;
  grid-row: 3;
}
