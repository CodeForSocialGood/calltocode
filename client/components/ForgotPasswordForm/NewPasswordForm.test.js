/* eslint-env jest */
import React from 'react'
import {shallow} from 'enzyme'

import {NewPasswordForm} from './NewPasswordForm'
let mockProps

beforeEach(() => {
  mockProps = {
    handleSubmit: () => {},
    onChangePassword: () => {},
    password: null
  }
})

describe('<NewPasswordForm />', () => {
  describe('render()', () => {
    describe('password of length 0', () => {
      beforeEach(() => { mockProps.password = '' })

      test('component renders correctly', () => {
        const component = shallow(<NewPasswordForm {...mockProps} />)
        expect(component).toMatchSnapshot()
      })
    })

    describe('password not of length 0', () => {
      beforeEach(() => { mockProps.password = 'fakepassword' })

      test('component renders correctly', () => {
        const component = shallow(<NewPasswordForm {...mockProps} />)
        expect(component).toMatchSnapshot()
      })
    })
  })
})
