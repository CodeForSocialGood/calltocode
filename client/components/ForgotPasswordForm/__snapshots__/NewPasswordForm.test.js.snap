// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NewPasswordForm /> render() password not of length 0 component renders correctly 1`] = `
<form
  className="form"
  onSubmit={[Function]}
>
  <h1
    className="h1"
  >
    New Password
  </h1>
  <h3
    className="h3"
  >
    Set your new password
  </h3>
  <Field
    component={[Function]}
    name="password"
    onChange={[Function]}
    validate={[Function]}
    value="fakepassword"
  />
  <button
    className="buttonSubmit"
    disabled={false}
    type="submit"
  >
    Change Password
  </button>
</form>
`;

exports[`<NewPasswordForm /> render() password of length 0 component renders correctly 1`] = `
<form
  className="form"
  onSubmit={[Function]}
>
  <h1
    className="h1"
  >
    New Password
  </h1>
  <h3
    className="h3"
  >
    Set your new password
  </h3>
  <Field
    component={[Function]}
    name="password"
    onChange={[Function]}
    validate={[Function]}
    value=""
  />
  <button
    className="buttonSubmit"
    disabled={true}
    type="submit"
  >
    Change Password
  </button>
</form>
`;
